batcher/batcher.mligo

```diff
@@ -1692,6 +1732,66 @@ let get_valid_swap_reduced
  | Some vswp -> vswp
  | None -> failwith swap_does_not_exist
 
+[@inline]
+let remove_orders_from_batch
+  (ots: ordertypes)
+  (batch: batch): batch =
+  let volumes = Batch_Utils.reduce_volumes ots batch.volumes in
+  { batch with volumes = volumes}
+
+[@inline]
+let remove_orders
+  (ot: ordertypes)
+  (batch:batch)
+  (batch_set: batch_set)
+  (storage: storage) =
+  let batch = remove_orders_from_batch ot batch in
+  let batches =  Big_map.update batch.batch_number (Some batch) batch_set.batches in
+  let batch_set = { batch_set with batches = batches } in
+  let storage = {storage with batch_set = batch_set} in
+  storage
+
+[@inline]
+let remove_order_types
+  (batch_number: nat)
+  (holder: address)
+  (bot: batch_ordertypes)
+  (storage:storage) : ordertypes * storage =
+  match Map.find_opt batch_number bot with
+  | None -> failwith no_orders_for_user_address
+  | Some ots -> let bot = Map.remove batch_number bot in
+                let ubots = Big_map.update holder (Some bot) storage.user_batch_ordertypes in
+                let storage = { storage with user_batch_ordertypes = ubots;} in
+                (ots,storage)
```

Comment:
Despite reading all of your PR 4 times, I don't even understand what this does. Do you remove the whole batch?

Reply:
Yes.  If there are no more holdings to redeem we remove the entire batch